amazon-personalize-ranking/src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -96,7 +101,14 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
 
             SearchHits personalizedHits = combineScores(hits, result);
             return personalizedHits;
-        } catch (Exception ex) {
+        } catch (AmazonServiceException e) {
+            logger.error("Exception while calling personalize campaign: {}", e.getMessage());
+            if (ACCESS_DENIED_EXCEPTION_ERROR_CODE.equals(e.getErrorCode())) {
+                throw new InvalidInputException(INSUFFCIENT_PUT_METRIC_PERMISSION_FORMAT);
+            }
```

Comment:
In order for OpenSearch to throw a 4xx error, I believe the thrown exception needs to be something that OpenSearch will recognize as a user input problem, like `IllegalArgumentException`.

OpenSearch doesn't know about AWS service exception classes.